### PR TITLE
replaced RocketChat with Discord

### LINKED ISSUE #947: Update chat links to go to Discord
Update any RocketChat links to point to Hyperledger Discord.

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>
Replace chat links with Discord link. 
I used the same link everywhere which is fine unless it's helpful to link directly to the besu channel

Fixes #947 
### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### CODE REVIEW

File: docs/HowTo/Use-Privacy/Use-GoQuorum-compatible-privacy.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ The following documentation explains [the transaction lifecycle] in GoQuorum-com
 !!! note
 
     If you intend to use this privacy mode for a production use case, please contact us on
-    [Rocket Chat](https://chat.hyperledger.org/channel/besu) or by [email](mailto:besu@lists.hyperledger.org).
+    [Hyperledger Besu Chat](https://discord.gg/hyperledger) or by [email](mailto:besu@lists.hyperledger.org).
```
Comment (alexandratran) [inline docs/HowTo/Use-Privacy/Use-GoQuorum-compatible-privacy.md:25]:
```suggestion
    [Hyperledger Besu chat](https://discord.gg/hyperledger) or by [email](mailto:besu@lists.hyperledger.org).
```

**Review by alexandratran — APPROVED** [review]
One suggestion

**Comment by macfarla** [comment]
@alexandratran I re-worded the chat links because in Discord, "chat" is short-hand for video chat channel. i.e. there is a "besu" channel which is the one we want, as well as a "besu chat" channel which is a voice channel which we will use rarely

File: README.md [hunk 2]
```diff
@@ -25,10 +25,10 @@ Having the following accounts is necessary for contributing code/issues to Besu:
 
 * If you want to contribute code, create a [Linux Foundation (LF) account] and a [GitHub account].
 * If you want to raise an issue, login to your [GitHub account].
-* [chat] also requires a [Linux Foundation (LF) account].
+* [Discord] also requires a Discord account.
```
Comment (alexandratran) [inline README.md:28]:
```suggestion
* [Hyperledger Discord] also requires a Discord account.
```

**Pull request merged**